lib/scenic/primitive/style/paint/color.ex

```diff
@@ -36,178 +38,55 @@ defmodule Scenic.Primitive.Style.Paint.Color do
-  def name_to_rgb(:medium_turquoise), do: {0x48, 0xD1, 0xCC}
-  def name_to_rgb(:medium_violet_red), do: {0xC7, 0x15, 0x85}
-  def name_to_rgb(:midnight_blue), do: {0x19, 0x19, 0x70}
-  def name_to_rgb(:mint_cream), do: {0xF5, 0xFF, 0xFA}
-  def name_to_rgb(:misty_rose), do: {0xFF, 0xE4, 0xE1}
-  def name_to_rgb(:moccasin), do: {0xFF, 0xE4, 0xB5}
-  def name_to_rgb(:navajo_white), do: {0xFF, 0xDE, 0xAD}
-  def name_to_rgb(:navy), do: {0x00, 0x00, 0x80}
-  def name_to_rgb(:old_lace), do: {0xFD, 0xF5, 0xE6}
-  def name_to_rgb(:olive), do: {0x80, 0x80, 0x00}
-  def name_to_rgb(:olive_drab), do: {0x6B, 0x8E, 0x23}
-  def name_to_rgb(:orange), do: {0xFF, 0xA5, 0x00}
-  def name_to_rgb(:orange_red), do: {0xFF, 0x45, 0x00}
-  def name_to_rgb(:orchid), do: {0xDA, 0x70, 0xD6}
-  def name_to_rgb(:pale_golden_rod), do: {0xEE, 0xE8, 0xAA}
-  def name_to_rgb(:pale_green), do: {0x98, 0xFB, 0x98}
-  def name_to_rgb(:pale_turquoise), do: {0xAF, 0xEE, 0xEE}
-  def name_to_rgb(:pale_violet_red), do: {0xDB, 0x70, 0x93}
-  def name_to_rgb(:papaya_whip), do: {0xFF, 0xEF, 0xD5}
-  def name_to_rgb(:peach_puff), do: {0xFF, 0xDA, 0xB9}
-  def name_to_rgb(:peru), do: {0xCD, 0x85, 0x3F}
-  def name_to_rgb(:pink), do: {0xFF, 0xC0, 0xCB}
-  def name_to_rgb(:plum), do: {0xDD, 0xA0, 0xDD}
-  def name_to_rgb(:powder_blue), do: {0xB0, 0xE0, 0xE6}
-  def name_to_rgb(:purple), do: {0x80, 0x00, 0x80}
-  def name_to_rgb(:rebecca_purple), do: {0x66, 0x33, 0x99}
-  def name_to_rgb(:red), do: {0xFF, 0x00, 0x00}
-  def name_to_rgb(:rosy_brown), do: {0xBC, 0x8F, 0x8F}
-  def name_to_rgb(:royal_blue), do: {0x41, 0x69, 0xE1}
-  def name_to_rgb(:saddle_brown), do: {0x8B, 0x45, 0x13}
-  def name_to_rgb(:salmon), do: {0xFA, 0x80, 0x72}
-  def name_to_rgb(:sandy_brown), do: {0xF4, 0xA4, 0x60}
-  def name_to_rgb(:sea_green), do: {0x2E, 0x8B, 0x57}
-  def name_to_rgb(:sea_shell), do: {0xFF, 0xF5, 0xEE}
-  def name_to_rgb(:sienna), do: {0xA0, 0x52, 0x2D}
-  def name_to_rgb(:silver), do: {0xC0, 0xC0, 0xC0}
-  def name_to_rgb(:sky_blue), do: {0x87, 0xCE, 0xEB}
-  def name_to_rgb(:slate_blue), do: {0x6A, 0x5A, 0xCD}
-  def name_to_rgb(:slate_gray), do: {0x70, 0x80, 0x90}
-  def name_to_rgb(:slate_grey), do: {0x70, 0x80, 0x90}
-  def name_to_rgb(:snow), do: {0xFF, 0xFA, 0xFA}
-  def name_to_rgb(:spring_green), do: {0x00, 0xFF, 0x7F}
-  def name_to_rgb(:steel_blue), do: {0x46, 0x82, 0xB4}
-  def name_to_rgb(:tan), do: {0xD2, 0xB4, 0x8C}
-  def name_to_rgb(:teal), do: {0x00, 0x80, 0x80}
-  def name_to_rgb(:thistle), do: {0xD8, 0xBF, 0xD8}
-  def name_to_rgb(:tomato), do: {0xFF, 0x63, 0x47}
-  def name_to_rgb(:turquoise), do: {0x40, 0xE0, 0xD0}
-  def name_to_rgb(:violet), do: {0xEE, 0x82, 0xEE}
-  def name_to_rgb(:wheat), do: {0xF5, 0xDE, 0xB3}
-  def name_to_rgb(:white), do: {0xFF, 0xFF, 0xFF}
-  def name_to_rgb(:white_smoke), do: {0xF5, 0xF5, 0xF5}
-  def name_to_rgb(:yellow), do: {0xFF, 0xFF, 0x00}
-  def name_to_rgb(:yellow_green), do: {0x9A, 0xCD, 0x32}
+  mapping =
```

Comment:
@milmazz is this done at the compiled time?

If the answer is not, then I wouldn't like this trade off.

Reply:
@yordis Yes, that's only for _compile time_ (at that should only occur when this file is updated or the `color_names.txt` is changed)

Reply:
Wouldn't be the same anyway btw? 

This list probably wouldn't change and to be honest I would prefer to maintain elixir code than reading files that people could get confused on how it would work.

IMO, I would prefer to just keep as it is right now, I don't see the benefits of it, but that is just an opinion anyway.

Reply:
I've been thinking about this.

@milmazz what is the reasoning for breaking the colors out into a separate file. I'm open minded to it, but don't know that I see the value. Also, I need to know if/how it affects performance.

I want to see some runtime benchmarking to prove what it does...

Otherwise I'm very happy with the tests. Can we get just those in a separate PR, then deal with the refact as we get data?

Reply:
> what is the reasoning for breaking the colors out into a separate file.

@boydm The reasoning it has to do more with maintenance and probably to improve the developer experience as I pointed out before to @yordis

> but don't know that I see the value

That's perfectly fine, I can delete this section if you want. Hopefully my previous comment convince you to keep it.

> I need to know if/how it affects performance.

I doesn't affect the _runtime performance_ at all, the transformation occurs only at _compile time_, which for this Module should occur only when:

1. This file (`color.ex`) is updated
1. When the file marked as [@external_resource](https://hexdocs.pm/elixir/Module.html#module-external_resource) is updated, that module attribute is used by Mix to know if the module needs recompilation.

To give you maybe another perspective: Elixir writes for you the 150 function clauses as result.

Just an example, I'll decompile the module to Erlang:

```console
λ mix decompile Scenic.Primitive.Style.Paint.Color --to erlang
λ perl -ne 'print if 132 .. 427' Elixir.Scenic.Primitive.Style.Paint.Color.erl
name_to_rgb(alice_blue) ->
    {240,248,255};
name_to_rgb(antique_white) ->
    {250,235,215};
name_to_rgb(aqua) ->
    {0,255,255};
name_to_rgb(aquamarine) ->
    {127,255,212};
name_to_rgb(azure) ->
    {240,255,255};
name_to_rgb(beige) ->
    {245,245,220};
name_to_rgb(bisque) ->
    {255,228,196};
name_to_rgb(black) ->
    {0,0,0};
name_to_rgb(blanched_almond) ->
    {255,235,205};
name_to_rgb(blue) ->
    {0,0,255};
name_to_rgb(blue_violet) ->
    {138,43,226};
name_to_rgb(brown) ->
    {165,42,42};
name_to_rgb(burly_wood) ->
    {222,184,135};
name_to_rgb(cadet_blue) ->
    {95,158,160};
name_to_rgb(chartreuse) ->
    {127,255,0};
name_to_rgb(chocolate) ->
    {210,105,30};
name_to_rgb(coral) ->
    {255,127,80};
name_to_rgb(cornflower_blue) ->
    {100,149,237};
name_to_rgb(cornsilk) ->
    {255,248,220};
name_to_rgb(crimson) ->
    {220,20,60};
name_to_rgb(cyan) ->
    {0,255,255};
name_to_rgb(dark_blue) ->
    {0,0,139};
name_to_rgb(dark_cyan) ->
    {0,139,139};
name_to_rgb(dark_golden_rod) ->
    {184,134,11};
name_to_rgb(dark_gray) ->
    {169,169,169};
name_to_rgb(dark_grey) ->
    {169,169,169};
name_to_rgb(dark_green) ->
    {0,100,0};
name_to_rgb(dark_khaki) ->
    {189,183,107};
name_to_rgb(dark_magenta) ->
    {139,0,139};
name_to_rgb(dark_olive_green) ->
    {85,107,47};
name_to_rgb(dark_orange) ->
    {255,140,0};
name_to_rgb(dark_orchid) ->
    {153,50,204};
name_to_rgb(dark_red) ->
    {139,0,0};
name_to_rgb(dark_salmon) ->
    {233,150,122};
name_to_rgb(dark_sea_green) ->
    {143,188,143};
name_to_rgb(dark_slate_blue) ->
    {72,61,139};
name_to_rgb(dark_slate_gray) ->
    {47,79,79};
name_to_rgb(dark_slate_grey) ->
    {47,79,79};
name_to_rgb(dark_turquoise) ->
    {0,206,209};
name_to_rgb(dark_violet) ->
    {148,0,211};
name_to_rgb(deep_pink) ->
    {255,20,147};
name_to_rgb(deep_sky_blue) ->
    {0,191,255};
name_to_rgb(dim_gray) ->
    {105,105,105};
name_to_rgb(dim_grey) ->
    {105,105,105};
name_to_rgb(dodger_blue) ->
    {30,144,255};
name_to_rgb(fire_brick) ->
    {178,34,34};
name_to_rgb(floral_white) ->
    {255,250,240};
name_to_rgb(forest_green) ->
    {34,139,34};
name_to_rgb(fuchsia) ->
    {255,0,255};
name_to_rgb(gainsboro) ->
    {220,220,220};
name_to_rgb(ghost_white) ->
    {248,248,255};
name_to_rgb(gold) ->
    {255,215,0};
name_to_rgb(golden_rod) ->
    {218,165,32};
name_to_rgb(gray) ->
    {128,128,128};
name_to_rgb(grey) ->
    {128,128,128};
name_to_rgb(green) ->
    {0,128,0};
name_to_rgb(green_yellow) ->
    {173,255,47};
name_to_rgb(honey_dew) ->
    {240,255,240};
name_to_rgb(hot_pink) ->
    {255,105,180};
name_to_rgb(indian_red) ->
    {205,92,92};
name_to_rgb(indigo) ->
    {75,0,130};
name_to_rgb(ivory) ->
    {255,255,240};
name_to_rgb(khaki) ->
    {240,230,140};
name_to_rgb(lavender) ->
    {230,230,250};
name_to_rgb(lavender_blush) ->
    {255,240,245};
name_to_rgb(lawn_green) ->
    {124,252,0};
name_to_rgb(lemon_chiffon) ->
    {255,250,205};
name_to_rgb(light_blue) ->
    {173,216,230};
name_to_rgb(light_coral) ->
    {240,128,128};
name_to_rgb(light_cyan) ->
    {224,255,255};
name_to_rgb(light_golden_rod_yellow) ->
    {250,250,210};
name_to_rgb(light_gray) ->
    {211,211,211};
name_to_rgb(light_grey) ->
    {211,211,211};
name_to_rgb(light_green) ->
    {144,238,144};
name_to_rgb(light_pink) ->
    {255,182,193};
name_to_rgb(light_salmon) ->
    {255,160,122};
name_to_rgb(light_sea_green) ->
    {32,178,170};
name_to_rgb(light_sky_blue) ->
    {135,206,250};
name_to_rgb(light_slate_gray) ->
    {119,136,153};
name_to_rgb(light_slate_grey) ->
    {119,136,153};
name_to_rgb(light_steel_blue) ->
    {176,196,222};
name_to_rgb(light_yellow) ->
    {255,255,224};
name_to_rgb(lime) ->
    {0,255,0};
name_to_rgb(lime_green) ->
    {50,205,50};
name_to_rgb(linen) ->
    {250,240,230};
name_to_rgb(magenta) ->
    {255,0,255};
name_to_rgb(maroon) ->
    {128,0,0};
name_to_rgb(medium_aqua_marine) ->
    {102,205,170};
name_to_rgb(medium_blue) ->
    {0,0,205};
name_to_rgb(medium_orchid) ->
    {186,85,211};
name_to_rgb(medium_purple) ->
    {147,112,219};
name_to_rgb(medium_sea_green) ->
    {60,179,113};
name_to_rgb(medium_slate_blue) ->
    {123,104,238};
name_to_rgb(medium_spring_green) ->
    {0,250,154};
name_to_rgb(medium_turquoise) ->
    {72,209,204};
name_to_rgb(medium_violet_red) ->
    {199,21,133};
name_to_rgb(midnight_blue) ->
    {25,25,112};
name_to_rgb(mint_cream) ->
    {245,255,250};
name_to_rgb(misty_rose) ->
    {255,228,225};
name_to_rgb(moccasin) ->
    {255,228,181};
name_to_rgb(navajo_white) ->
    {255,222,173};
name_to_rgb(navy) ->
    {0,0,128};
name_to_rgb(old_lace) ->
    {253,245,230};
name_to_rgb(olive) ->
    {128,128,0};
name_to_rgb(olive_drab) ->
    {107,142,35};
name_to_rgb(orange) ->
    {255,165,0};
name_to_rgb(orange_red) ->
    {255,69,0};
name_to_rgb(orchid) ->
    {218,112,214};
name_to_rgb(pale_golden_rod) ->
    {238,232,170};
name_to_rgb(pale_green) ->
    {152,251,152};
name_to_rgb(pale_turquoise) ->
    {175,238,238};
name_to_rgb(pale_violet_red) ->
    {219,112,147};
name_to_rgb(papaya_whip) ->
    {255,239,213};
name_to_rgb(peach_puff) ->
    {255,218,185};
name_to_rgb(peru) ->
    {205,133,63};
name_to_rgb(pink) ->
    {255,192,203};
name_to_rgb(plum) ->
    {221,160,221};
name_to_rgb(powder_blue) ->
    {176,224,230};
name_to_rgb(purple) ->
    {128,0,128};
name_to_rgb(rebecca_purple) ->
    {102,51,153};
name_to_rgb(red) ->
    {255,0,0};
name_to_rgb(rosy_brown) ->
    {188,143,143};
name_to_rgb(royal_blue) ->
    {65,105,225};
name_to_rgb(saddle_brown) ->
    {139,69,19};
name_to_rgb(salmon) ->
    {250,128,114};
name_to_rgb(sandy_brown) ->
    {244,164,96};
name_to_rgb(sea_green) ->
    {46,139,87};
name_to_rgb(sea_shell) ->
    {255,245,238};
name_to_rgb(sienna) ->
    {160,82,45};
name_to_rgb(silver) ->
    {192,192,192};
name_to_rgb(sky_blue) ->
    {135,206,235};
name_to_rgb(slate_blue) ->
    {106,90,205};
name_to_rgb(slate_gray) ->
    {112,128,144};
name_to_rgb(slate_grey) ->
    {112,128,144};
name_to_rgb(snow) ->
    {255,250,250};
name_to_rgb(spring_green) ->
    {0,255,127};
name_to_rgb(steel_blue) ->
    {70,130,180};
name_to_rgb(tan) ->
    {210,180,140};
name_to_rgb(teal) ->
    {0,128,128};
name_to_rgb(thistle) ->
    {216,191,216};
name_to_rgb(tomato) ->
    {255,99,71};
name_to_rgb(turquoise) ->
    {64,224,208};
name_to_rgb(violet) ->
    {238,130,238};
name_to_rgb(wheat) ->
    {245,222,179};
name_to_rgb(white) ->
    {255,255,255};
name_to_rgb(white_smoke) ->
    {245,245,245};
name_to_rgb(yellow) ->
    {255,255,0};
name_to_rgb(yellow_green) ->
    {154,205,50}.
```

Hope that helps. Please let me know what do you think or let me know if you need more information.

Reply:
I think I just hit the use case that brings me over to taking this. I'm working on the docs and would like to get the named colors and their values as a list to generate the docs...

I'll play with that.